openlibrary/plugins/upstream/account.py

```diff
@@ -244,6 +244,8 @@ def POST(self):
                 InternetArchiveAccount.create(
                     screenname=f.username.value, email=f.email.value, password=f.password.value,
                     notifications=notifications, verified=False, retries=USERNAME_RETRIES)
+                user = OpenLibraryAccount.get(email=f.email.value)
```

Comment:
This looks close -- I'm not positive if, at this point, the Open Library account has been created. I think at first, until activation, **only** the Archive.org account exists.

Their first login / post-activation is what create their account. Let's double check this because I get the feeling this code is correct, just maybe in the wrong place. :)

Reply:
Okay :)

Reply:
@mekarpeles Do we know somewhere that the user is logging in into his account for the first time, After verification?

Reply:
Confirmed this code won't work as is:
```
Traceback (most recent call last):
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/web/application.py", line 239, in process
    return self.handle()
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/web/application.py", line 230, in handle
    return self._delegate(fn, self.fvars, args)
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/web/application.py", line 462, in _delegate
    return handle_class(cls)
  File "/opt/openlibrary/venv/local/lib/python2.7/site-packages/web/application.py", line 438, in handle_class
    return tocall(*args)
  File "/opt/openlibrary/deploys/openlibrary/openlibrary/infogami/utils/app.py", line 189, in <lambda>
    HEAD = GET = POST = PUT = DELETE = lambda self: delegate()
  File "/opt/openlibrary/deploys/openlibrary/openlibrary/infogami/utils/app.py", line 208, in delegate
    return getattr(cls(), method)(*args)
  File "/opt/openlibrary/deploys/openlibrary/openlibrary/openlibrary/plugins/upstream/account.py", line 248, in POST
    user.save_preferences({'public_readlog':'yes'})
AttributeError: 'NoneType' object has no attribute 'save_preferences'
```

Reply:
plugins/upstream/account.py `account_create` is what we want

Reply:
audit_accounts is what's called when the user logs in (when its their first time, the linking logic occurs and we may set `save_preferences`

Reply:
Directly after https://github.com/internetarchive/openlibrary/blob/6846e736a8994f11bd980d82598c53bbd0d92b3f/openlibrary/accounts/model.py#L698-L701 all we need is `ol_account.save_preferences({'public_readlog':'yes'})`

Reply:
Thanks @mekarpeles for pointing out the correct file to place the code :) Pushed the new changes and its ready to be merged